### ISSUE TITLE
New command line args are misleading (sort and likely other commands; ver 1.2)

### ISSUE THREAD

**Issue by balwierz** [samtools/samtools#349]
It seems like samtools don't obey command line options. See the 3 examples below
````
$ samtools sort foo.bam
Usage: samtools sort [options...] [in.bam]
  -l INT     Set compression level, from 0 (uncompressed) to 9 (best)
  -m INT     Set maximum memory per thread; suffix K/M/G recognized [768M]
  -n         Sort by read name
  -o FILE    Write final output to FILE rather than standard output
  -O FORMAT  Write output as FORMAT ('sam'/'bam'/'cram')   (either -O or
  -T PREFIX  Write temporary files to PREFIX.nnnn.bam       -T is required)
  -@ INT     Set number of sorting and compression threads [1]

Legacy usage: samtools sort [options...] <in.bam> <out.prefix>
Options:
  -f         Use <out.prefix> as full final filename rather than prefix
  -o         Write final output to stdout rather than <out.prefix>.bam
  -l,m,n,@   Similar to corresponding options above
```
But this is exactly how I ran it.

Moreover after running
```
$ samtools sort -o bar.bam foo.bam
[E::hts_open] fail to open file 'bar.bam'
[bam_sort_core] fail to open file bar.bam
```

And after running
`$ samtools sort -@ 4 foo.bam -o bar.bam`
there is no result file. But it was sorting using threads, but never the generated the final bar.bam.

All of the above examples are valid according to the usage info displayed by samtools. For now I am sticking to the "legacy" options. Please fix it.

**Comment by balwierz** [comment]
I guess it is because of
```c++
int bam_sort(int argc, char *argv[])
{
 ... [SKIP]
modern = 0;
for (i = 1; i < argc; ++i)
if (argv[i][0] == '-' && strpbrk(argv[i], "OT")) { modern = 1; break; }
...
```
(And it is mentioned to use either -O or -T, but in the wrong place.)
The usage should read:
`Usage: samtools sort <-T PREFIX | -O FORMAT> [options...] [in.bam]`

**Comment by Loganskier** [comment]
I am having the same problem.  
The samtools sort functioned just fine for the following
samtools sort -o DP90_04pes_output.sam -O sam -T DP90_04_tmp -@ 8 DP90_04pe_output.sam

when I ran 

samtools sort -o DP90_24pes_output.sam -O sam -T DP90_24_tmp -@ 8 DP90_24pe_output.sam
I got  [E::hts_open] fail to open file 'DP90_24pes_output.sam'

I have tried many different arrangements and nothing seems to work. please fix!

**Comment by SamStudio8** [comment]
@Loganskier In your example you seem to be passing your desired output file as your input also?

**Comment by SamStudio8** [comment]
@Loganskier Sorry I missed the additional `s` in the filename for `-o`, apparently I can't read. See #418 also, I've also proposed a fix as #425. This mess will be fixed in the next version of `samtools`.

**Comment by jmarshall** [comment]
Fixed by removing the legacy `<out.prefix>` invocation style; this removal enables `-T` to be optional.
